src/main/java/seedu/address/model/CommandRecord.java

```diff
@@ -27,16 +25,23 @@ public CommandRecord(Integer index, String commandString, CommandType commandTyp
         this.commandType = commandType;
     }
 
-    public CommandRecord(CommandType commandType) throws AlfredModelHistoryException {
-        if (commandType.equals(CommandType.END)) {
-            this.commandString = "*: Initialised State. Cannot undo.";
-            this.index = null;
-            this.commandType = CommandType.END;
-        } else {
-            throw new AlfredModelHistoryException("Previous commands cannot be properly initialised");
-        }
+    public static CommandRecord getUndoEndPoint() {
+        return new CommandRecord(null,
+                   "=============<< Cannot Undo Beyond This Point >>=============",
+                                 CommandType.END);
+    }
+
+    public static CommandRecord getRedoEndPoint() {
+        return new CommandRecord(null,
```

Comment:
Why not make these marker command records static/final?

Reply:
I was afraid the fields within the static CommandRecord object might be inadvertently edited

Reply:
Hmmm actually why not just make the marker (endpoint) command records a dummy record. So you pop until you see it